### DISCUSSION TITLE
Choose orientation of vertical textbox and other vertical supported rotated widgets

### DISCUSSION THREAD

**Discussion by TheHighGroundWins** [qtile/qtile#5127] [Ideas]
Many of the widgets support vertical bar orientations which is great but they are either automatically 90 degrees on left bar, and -90 degrees on right bar.

Some writing systems (like Mongolian Script) are vertical but only read in one orientation. Which means its only readable in right bar, which is confusing since the language is read left to right.
![20241220_17h41m01s_grim](https://github.com/user-attachments/assets/252d9ca8-bd2a-456f-8170-4539597a2e21)

I have been using custom widget by extending existing widgets and overriding the draw function to rotate in the other direction.

Maybe for now just do textbox, and later add support for other widgets

For example for textbox:

When it is normally
```
        if not self.bar.horizontal:
            # Left bar reads bottom to top
            if self.bar.screen.left is self.bar:
                self.drawer.ctx.rotate(-90 * math.pi / 180.0)
                self.drawer.ctx.translate(-self.length, 0)

            # Right bar is top to bottom
            else:
                self.drawer.ctx.translate(self.bar.width, 0)
                self.drawer.ctx.rotate(90 * math.pi / 180.0)

```
a change could be:
```
        if not self.bar.horizontal:
            if not top-to-bottom
                self.drawer.ctx.rotate(-90 * math.pi / 180.0)
                self.drawer.ctx.translate(-self.length, 0)

            # for top-to-bottom
            else:
                self.drawer.ctx.translate(self.bar.width, 0)
                self.drawer.ctx.rotate(90 * math.pi / 180.0)

```
the name of the parameter, could be top-to-bottom as its easier to understand than bottom-to-top.
And keep the orientations of bottom-to-top on the top, in order to create minimal changes to source code, without having to rearrange too much.

This change is easy enough that I could at least do a pull request for textbox, without needing someone without extensive knowledge.

**Comment by elParaguayo** [comment]
This change would need to be in the `base._TextBox` class.

I'm happy for you to submit a pull request for this. However, if the wiring script is vertical, surely you don't want to rotate the text at all?

**Reply by TheHighGroundWins** [reply]
The way the Unicode and font itself is designed, is made to have all the characters sideways when horizontal like this: ᠮᠣᠩᠭᠣᠯ.

And then rotated -90 degrees to have the correct upright orientation.

It was designed this way so that all the underline, and other text effects would still look correct without any changes in text handling.

**Comment by TheHighGroundWins** [comment]
Managed to have it so that, it would have to be overriden. So that the default behavior doesn't unchange unless overriden.
